Multitouch Support/Native/VoodooI2CNativeEngine.cpp

```diff
@@ -92,6 +91,14 @@ MultitouchReturn VoodooI2CNativeEngine::handleInterruptReport(VoodooI2CMultitouc
         
     super::messageClient(kIOMessageVoodooInputMessage, voodooInputInstance, &message, sizeof(VoodooInputEvent));
     
+    if (!isForceClickEnabled()) {
```

Comment:
```suggestion
    if (firstTransducer->physical_button.value() != firstTransducer->physical_button.last.value && !isForceClickEnabled()) {
```

Reply:
@Goshin what's your reasoning for the above change?

Reply:
As suggested by @ben9923, this is to track the stage change and send the report only when the button is pressed and released.

Reply:
Tested with this change, buttons don't work at all. Don't know why, maybe I'll try to debug it later.

Reply:
`VoodooI2CMultitouchHIDEventDriver::setButtonState` will be called twice in each event as there are multiple buttons, making `.last` value pretty much invalid.

Reply:
> VoodooI2CMultitouchHIDEventDriver::setButtonState will be called twice in each event as there are multiple buttons, making .last value pretty much invalid.

So an easy way (might be a little bit ugly) to fix this would be to update the last value only when the value really changes? Something like:

```diff
class TimeTrackedValue {
public:
    ValueWithTime current;
    ValueWithTime last;
...
     void update(UInt16 value, AbsoluteTime timestamp) {
-        last = current;
+        if (current.value != value) {
+            last = current;
+        }
         current.value = value;
         current.timestamp = timestamp;
     }
...
};
```

otherwise, the other ways I can come up with are to refactor `::setButtonState()` to accept two button state values at a time, or to bypass it and manually manipulate the value of `transducer->physical_button`, which is ugly.

Reply:
@Goshin Refactoring (or rather adding `::setButtonMultiState`/whatever) seemed like the proper solution to me. Then I also figured we might have a problem when input is given in multiple digitizer reports.
In such cases state would be updated multiple times anyway (`::handleDigitizerReport` called for each report). Maybe parsing button elements should only be done in first/last report (I wonder how it works in Linux).

Am I right here? CC @kprinssu @usr-sse2

Reply:
> Refactoring (or rather adding ::setButtonMultiState/whatever) seemed like the proper solution to me. Then I also figured we might have a problem when input is given in multiple digitizer reports.
In such cases state would be updated multiple times anyway

@ben9923 I agree. How about keeping a `lastPhysicalButtonValue` var in `VoodooI2CNativeEngine`? Will it make it a lot easier? (and a little bit uglier of course.) The last value will be updated only when sending a touch report to VoodooInput, so it can work around those mentioned problems.

Reply:
@Goshin Thanks for the confirmation :)
I think keeping the last value should rather be done on the VoodooI2CHID side, as the reported value to the native engine already contains a 'last' value, making it really ambiguous.
IMO another `DigitiserTransducerButtonState` for the button should be a part of the `digitiser`, it will then be properly updated once creating the `VoodooI2CMultitouchEvent` and put into the first transducer.

Also, ideally the button shouldn't be a part of the first transducer but rather a separate filed of the report.
Not sure why it is like that now.

Reply:
We can address this in a follow-up PR. It's good as-is.